design/proposals/load_policy/load_policy_validation.md

```diff
@@ -0,0 +1,148 @@
+2. Conjur admin load a policy of Host to be authenticated with JWT authenticator
+3. New policy validation layer is triggered and validate if the host is configured correctly 
+4. Host authenticate using JWT authenticator, due to above validation there are no configuration errors at the authenticate request stage
+
+### Sequence Diagrams
+![Load policy validation sequence diagram](load-policy-validation-sequence-diagram.png)
+
+#### Sequence Diagrams - additional info
+1. Policy validation entry point:
+As described in the diagram the validation will be triggered as the final new stage of the current logic (read below),
+   which will call to a new command class `ValidatePolicyConstraints` with tne input of the policy entities 
+   records (see example in the end the of the diagram) from the user body request.
+   
+Load policy current logic - 
+   for the load policy rest requests put/post/patch the current logic in high level is:
+    The current logic: 
+       
+- policy syntax check
+- load records from policy 
+- create temp schema
+- merge new temp schema to main schema 
+- the validation logic will start after the schema merged to main
+
+2. new ValidatePolicyConstraints command class - is responsible to validate all the policy entities 
+   (host,user,webservice) which were sent by the user in load policy request.
+   
+    Note: in this design we will validate only hosts 
+   
+3. new CreatePolicyEntityValidatorsFactory command class - is responsible to create a list of validators as the amount 
+   of the different policy entities types in the records input.
+   
+    Note: in this design we will create only `ValidateHostConstraints` with all the hosts ids in his belly 
+   
+4. new ValidateHostConstraints command class - is responsible to validate a list of host ids which are initialized in his creation.
+   the flow is to fetch the host annotations from the DB (from the main schema after the merge) and to 
+   use the `CreateHostValidatorsFactory` to create hosts authenticators validators.
+   
+5. new CreateHostValidatorsFactory command class - is responsible to create a list of validators as the amount
+   of the different hosts authenticators types
+   
+    Note: 
+    - in this design we will create only `ValidateJwtHostConstraints` which is validator for hosts with JWT annotations
+    - this class can be refactored to use reflection if in the future we will want the separate the authenticator logic 
+    to a different gems 
+      
+6. new GetJwtHosts command class - is responsible to get a lists of host and annotations and to return only the hosts wjich are 
+related the JWT authenticator
+   
+7. new ValidateJwtHostConstraints command class - is responsible to to perform the validation logic as written in the requirements,
+for all annotations have values.
+   
+    Note: we will refactor the current code of the authentication flow to use this new command class as well 
+   
+
+   
+
+## Performance
+TBD
```

Comment:
@nessiLahav I know you said several times that this has performance plenty that we will need to understand later on

Reply:
I updated this section but still do not know to say how big will be the  performance plenty